documentation/authentication.md

```diff
@@ -0,0 +1,135 @@
+Authentication
+================
+
+Authenticating requests to `dotnet-monitor` helps protect sensitive diagnostic artifacts from unauthorized users and lower privileged processes. `dotnet-monitor` can be configured to use either [Windows Authentication](#windows-authentication) or via an [API Token](#api-token-authentication). It is possible, although strongly not recommended, to [disable authentication](#disabling-authentication).
```

Comment:
We should also mention we only perform authentication on non-metrics endpoints

Reply:
Is it true that we only authenticate on a secure channel? I mentioned this down on Line 14, but I believe the behavior is to emit a warning when TLS is not enabled. @wiktor/@jander - could one of you confirm?

Reply:
If not, let's add something in documentation to discourage using authentication over an insecure channel.

Reply:
We do allow authentication thought plain http, but we log a warning message when configured that way:

`WARNING: Authentication is enabled over insecure http transport. This can pose a security risk and is not intended for production environments.`